docs/rfcs/xxx-pretranspile.md

```diff
@@ -0,0 +1,44 @@
+# Pre-transpiled Atom packages
+
+## Status
+
+Proposed
+
+## Summary
+
+This feature will enable package authors to use conventional npm tooling and package.json conventions to take advantage of JavaScript transpilers like Babel or TypeScript.
+
+## Motivation
+
+Transpiling packages on _publish_ rather than _load_ will have great benefits for package authors:
+
+* Standard `npm` tooling like `prepare` scripts will work for apm packages exactly as they work for npm packages. This will remove the need for custom transpiler pipeline modules like [atom-babel6-transpiler](https://github.com/atom/atom-babel6-transpiler) or [atom-typescript-transpiler](https://github.com/smhxx/atom-ts-transpiler) with their own, independent documentation, configuration and setup.
+* Packages can move transpiler-related dependencies to `devDependencies` and trim installation bloat substantially. (as a data point, the TypeScript compiler is 30MB.)
+* First-time package load will no longer take a hit from transpiling all of the source into the cache. All package loads will benefit from no longer needing to check and side-load from the compiler cache.
+
+## Explanation
+
+### Package publishing
+
+During the `apm publish` call, apm will invoke [`npm pack`](https://docs.npmjs.com/cli/pack) to run all standard npm lifecycle hooks and prepare a `.tar.gz` file. apm then creates a release associated with the created tag and uploads the `.tar.gz` file as an attachment called `package.tar.gz`.
```

Comment:
Will more "manual" workflows still be supported? For example right now if you want to create a signed release tag you need to do everything manually.

I'm guessing that as long as the `package.tar.gz` is attached and the version fits the appropriate format this will be the case, but it might be better to explicitly state this.

Reply:
You mean doing the git steps manually and using `apm publish --tag` to upload it to atom.io? Yeah, that should still work fine. I'll add a note 🙇

Reply:
> You mean doing the git steps manually and using apm publish --tag to upload it to atom.io?

Yep, [exactly](https://gist.github.com/Arcanemagus/d068e3f990ca4b38fa69a1425e3a76fd).